### PR TITLE
Converted to React

### PR DESCRIPTION
**Pull request by HadarRose**
replaced basicRequests.js with multiple React components

### CODE REVIEW

File: src/js/timeline.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            error: null,
+        });
+        this.timelineService.makeRequest();
+    }
+
+    render() {
+        if(this.state.isLoaded && this.state.error){ // if loaded and an error, render error message
+            console.error(this.state.error);
+            return(
+                <div className="timeline-container">
```
Comment (kev-chan) [inline src/js/timeline.js]: timeline-container is still duplicated

File: src/js/timelineService.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+export default class TimelineService{
```
Comment (kev-chan) [inline src/js/timelineService.js:1]: how re-usable is this class?

**Review by kev-chan — CHANGES_REQUESTED** [review]

**Comment by HadarRose** [comment]
@kev-chan I pushed a new version. Took care of the duplication, and made it so that the service calls "timeline.updateState" instead, which means it could be used by any class with an updateState method in the way that updateState is implemented. (as opposed to setting the state from the service, since different callers may have different states)

File: src/js/timelinePage.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    getTimeline(){ // calls the timeline component's requestTimeline
+        this.refs.timeline.requestTimeline();
+    }
+
+    render(){
+        return(
+            <div className="main-container">
+                <div className="title-div">Lab for Hadar</div>
+                <button id="timeline-button" onClick={() => this.getTimeline()} >Get Timeline</button> 
+                <div className="timeline-container">
```
Comment (kev-chan) [inline src/js/timelinePage.js]: are you expecting this to be part of timeline if someone else was to just render `<Timeline ...>?

File: src/js/timelineService.js (hunk 2 above)
Comment (kev-chan) [reply src/js/timelineService.js:1]: update to use function

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/js/timelineService.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    constructor(func){
```
Comment (kev-chan) [inline src/js/timelineService.js]: call it `callback` rather than func.  It's the actual technical term for it

**Pull request merged**